File: java/strategy/ActionGame/client/Character.java
package client;

import behavior.WeaponBehavior;

public class Character {

    WeaponBehavior weaponBehavior;

    public void fight() {
        weaponBehavior.useWeapon();
    }

    public void setWeapon(WeaponBehavior weaponBehavior) {
        this.weaponBehavior = weaponBehavior;
    }
}

File: kotlin/strategy/ActionGame/client/Character.kt
package client

import behavior.WeaponBehavior

open class Character {

    protected open var weaponBehavior: WeaponBehavior? = null

    fun fight() {
        weaponBehavior!!.useWeapon()
    }

    fun setWeapon(newWeapon: WeaponBehavior) {
        weaponBehavior = newWeapon
    }

}